pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -930,3 +956,47 @@ func GetSelectorRegexMap(l7 policy.L7DataMap) (CachedSelectorREEntry, error) {
 
 	return newRE, nil
 }
+
+// Removes the last trailing dot in case its an fqdn
+// TODO move to the proper location
```

Comment:
These can probably be moved to the `pkg/fqdn/matchpattern` pkg as there exists similar code to this there.

Reply:
Sounds good, thanks!